Wrappers/Python/cil/optimisation/functions/TotalVariation.py

```diff
@@ -195,12 +198,18 @@ def proximal(self, x, tau, out = None):
             self._domain = x
         
         # initialise
-        t = 1        
-        tmp_p = self.gradient.range_geometry().allocate(0)  
-        tmp_q = tmp_p.copy()
-        tmp_x = self.gradient.domain_geometry().allocate(0)     
-        p1 = self.gradient.range_geometry().allocate(0)
-        
+        t = 1
+        if not self.warmstart:      
+            self.p1 = self.gradient.range_geometry().allocate(0)
+        else:
+            if not self.hasstarted:  
+                self.p1 = self.gradient.range_geometry().allocate(0)
```

Comment:
because `p1` is member of the class, there is no difference between the warmstart and no warmstart implementation wrt memory usage: both allocate and store the `p1` buffer.